File: web/backend/src/services/redis.js
const redis = require('redis');

const client = redis.createClient({
  url: process.env.REDIS_URL,
});
client.connect().catch(console.error);

const cache = async (key, ttl, fetchData) => {
  const cached = await client.get(key);
  if (cached) return JSON.parse(cached);

  const data = await fetchData();
  await client.setEx(key, ttl, JSON.stringify(data));
  return data;
};

module.exports = { redisClient: client, cache };

File: web/backend/src/services/redis.py
import redis
import os

REDIS_URL = os.getenv("REDIS_URL", "redis://redis:6379")
redis_client = redis.Redis.from_url(REDIS_URL)

def cache_data(key: str, value: str, expire: int = 3600):
    redis_client.setex(key, expire, value)

def get_cached_data(key: str):
    return redis_client.get(key)